src/Elm/Parser/State.elm

```diff
@@ -15,9 +29,20 @@ emptyState =
     State
         { indents = []
         , comments = []
+        , importOrDeclarationParsed = False
         }
 
 
+parsedImportOrDeclaration : State -> State
+parsedImportOrDeclaration (State state) =
+    State { state | importOrDeclarationParsed = True }
+
+
+checkParsedImportOrDeclaration : State -> Bool
+checkParsedImportOrDeclaration (State { importOrDeclarationParsed }) =
```

Comment:
I don't really like the function names I came up with here, suggestions welcome.